wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -0,0 +1,2601 @@
+		WP_MySQL_Lexer::MULTIPOINT_SYMBOL         => 'TEXT',
+		WP_MySQL_Lexer::MULTILINESTRING_SYMBOL    => 'TEXT',
+		WP_MySQL_Lexer::MULTIPOLYGON_SYMBOL       => 'TEXT',
+		WP_MySQL_Lexer::GEOMCOLLECTION_SYMBOL     => 'TEXT',
+		WP_MySQL_Lexer::GEOMETRYCOLLECTION_SYMBOL => 'TEXT',
+
+		// SERIAL, SET, and JSON types are handled in the translation process.
+	);
+
+	/**
+	 * A map of normalized MySQL data types to SQLite data types.
+	 *
+	 * This is used to generate SQLite CREATE TABLE statements from the MySQL
+	 * INFORMATION_SCHEMA tables. They keys are MySQL data types normalized
+	 * as they appear in the INFORMATION_SCHEMA. Values are SQLite data types.
+	 */
+	const DATA_TYPE_STRING_MAP = array(
+		// Numeric data types:
+		'bit'                => 'INTEGER',
+		'bool'               => 'INTEGER',
+		'boolean'            => 'INTEGER',
+		'tinyint'            => 'INTEGER',
+		'smallint'           => 'INTEGER',
+		'mediumint'          => 'INTEGER',
+		'int'                => 'INTEGER',
+		'integer'            => 'INTEGER',
+		'bigint'             => 'INTEGER',
+		'float'              => 'REAL',
+		'double'             => 'REAL',
+		'real'               => 'REAL',
+		'decimal'            => 'REAL',
+		'dec'                => 'REAL',
+		'fixed'              => 'REAL',
+		'numeric'            => 'REAL',
+
+		// String data types:
+		'char'               => 'TEXT',
+		'varchar'            => 'TEXT',
+		'nchar'              => 'TEXT',
+		'nvarchar'           => 'TEXT',
+		'tinytext'           => 'TEXT',
+		'text'               => 'TEXT',
+		'mediumtext'         => 'TEXT',
+		'longtext'           => 'TEXT',
+		'enum'               => 'TEXT',
+		'set'                => 'TEXT',
+		'json'               => 'TEXT',
+
+		// Date and time data types:
+		'date'               => 'TEXT',
+		'time'               => 'TEXT',
+		'datetime'           => 'TEXT',
+		'timestamp'          => 'TEXT',
+		'year'               => 'TEXT',
+
+		// Binary data types:
+		'binary'             => 'INTEGER',
+		'varbinary'          => 'BLOB',
+		'tinyblob'           => 'BLOB',
+		'blob'               => 'BLOB',
+		'mediumblob'         => 'BLOB',
+		'longblob'           => 'BLOB',
+
+		// Spatial data types:
+		'geometry'           => 'TEXT',
+		'point'              => 'TEXT',
+		'linestring'         => 'TEXT',
+		'polygon'            => 'TEXT',
+		'multipoint'         => 'TEXT',
+		'multilinestring'    => 'TEXT',
+		'multipolygon'       => 'TEXT',
+		'geomcollection'     => 'TEXT',
+		'geometrycollection' => 'TEXT',
+	);
+
+	/**
+	 * A map of MySQL to SQLite date format translation.
+	 *
+	 * It maps MySQL DATE_FORMAT() formats to SQLite STRFTIME() formats.
+	 *
+	 * For MySQL formats, see:
+	 *   https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_date-format
+	 *
+	 * For SQLite formats, see:
+	 *   https://www.sqlite.org/lang_datefunc.html
+	 *   https://strftime.org/
+	 */
+	const DATE_FORMAT_TO_STRFTIME_MAP = array(
```

Comment:
Nitpick: I noticed I needed to read the comment to understand the mapping, we could encode more information in the constant name:

```suggestion
		const MYSQL_DATE_FORMAT_TO_SQLITE_STRFTIME_MAP = array(
```

Reply:
Good point! Fixed.